pkg/outputs/kafka_output/kafka_output.go

```diff
@@ -280,16 +296,125 @@ func (k *kafkaOutput) RegisterMetrics(reg *prometheus.Registry) {
 }
 
 func (k *kafkaOutput) worker(ctx context.Context, idx int, config *sarama.Config) {
+	if k.cfg.SyncProducer {
+		k.syncProducerWorker(ctx, idx, config)
+	}
+	k.asyncProducerWorker(ctx, idx, config)
```

Comment:
Should there be an `else` here?

Reply:
Right, those worker functions are blocking calls. I should add a return after the syncProducerWorker() call. I try to avoid `else` statements.